src/engine/SCons/Script/Main.py

```diff
@@ -714,11 +714,20 @@ def _load_site_scons_dir(topdir, site_dir_name=None):
     site_init_file = os.path.join(site_dir, site_init_filename)
     site_tools_dir = os.path.join(site_dir, site_tools_dirname)
     if os.path.exists(site_init_file):
-        import imp, re
+        if PY3:
+            import importlib
+        else:
+            import imp
+        import re
+
         try:
             try:
-                fp, pathname, description = imp.find_module(site_init_modname,
-                                                            [site_dir])
+                if PY3:
+                    spec = importlib.util.find_spec(site_init_modname)
+                    fp = open(spec.origin, 'r')
```

Comment:
how would this find the site_dir?

Reply:
existing code adds it to `sys.path` above. it seems to work for the tests at least. importlib method doesn't have way to pass same arg.